### DISCUSSION TITLE
New to SSR and Nextjs.

### DISCUSSION THREAD

**Discussion by juanUrdaneta** [vercel/next.js#15726] [Help]
Hello, I'm new to SSR and I wanted to learn Nextjs so I decided to use the pokeapi to make a simple pokedex that works this way: 

1. search for pokemon
2. see a list that consist on the pokemon you search for and type like pokemons 
3. click on one pokemon and enter a detailed page with it's data. 

The app works by me making a request to the pokeapi each time the users search for a pokemon, I have zero data stored on the app. All the data needed for the details is fetched on the first search

Two issues I have encountered that I have not been able to sort reading the docs:

1. I'm not entirely sure how to pass the already fetched pokemon data that is already on the search result (index) page to the dynamic route page for previewing the detailed information. I can pass a couple of query parameters but I'm not sure how to get the entire object to the detailes page. 

2. Before making this work I noticed that the page with the search result is not cached so each time I go back from the details page (the one that I mentioned above that is currently empty) the search page is empty again as if I haven't search anything. 

I'm sorry if I'm too much of a noob but I'd like to ask: What am I missing here? What should I be looking into?

**Comment by arso1er** [comment]
Hi. I'm no expert, but this is what I think.

1. In your dynamic detail page, fetch the pokemon details with `getStaticProps` or `getServerSideProps` ([docs here](https://nextjs.org/docs/basic-features/data-fetching)). If you already have all details on the search result page, you could pass them to the detail page using the context api (example [here](https://github.com/vercel/next.js/tree/canary/examples/with-context-api)) but that would be bad for SEO.

2. For caching, you could try [SWR](https://nextjs.org/docs/basic-features/data-fetching#swr).

**Reply by juanUrdaneta** [reply]
Thanks! I'm already using serversidprops but caching is something I didnt knew I could do.
